pkg/jobrunaggregator/jobruntestcaseanalyzer/cmd.go

```diff
@@ -0,0 +1,241 @@
+	usage := fmt.Sprintf("mutually exclusive to --payload-tag.  Matches the .label[%s] on the prowjob, which is a UID", jobrunaggregatorlib.PayloadInvocationIDLabel)
+	fs.StringVar(&f.PayloadInvocationID, "payload-invocation-id", f.PayloadInvocationID, usage)
+
+	fs.StringVar(&f.WorkingDir, "working-dir", f.WorkingDir, "The directory to store caches, output, and the like.")
+	fs.DurationVar(&f.Timeout, "timeout", f.Timeout, "Time to wait for analyzing job to complete.")
+	fs.Var(&jobGcsPrefixSlice{&f.JobGcsPrefixes}, "explicit-gcs-prefixes", "a list of gcs prefixes for jobs created for payload. Only used by per PR payload promotion jobs. The format is comma-separated elements, each consisting of job name and gcs prefix separate by tilde, like openshift-machine-config-operator-3028-ci-4.11-e2e-aws-ovn-upgrade~logs/openshift-machine-config-operator-3028-ci-4.11-e2e-aws-ovn-upgrade")
+}
+
+func NewJobRunsTestCaseAnalyzerCommand() *cobra.Command {
+	f := NewJobRunsTestCaseAnalyzerFlags()
+
+	cmd := &cobra.Command{
+		Use:          "analyze-required-install",
+		Long:         `Analyze installation status of selected jobs to make sure they meet minimum criteria specified`,
+		SilenceUsage: true,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			if err := f.Validate(); err != nil {
+				logrus.WithError(err).Fatal("Flags are invalid")
+			}
+			o, err := f.ToOptions(ctx)
+			if err != nil {
+				logrus.WithError(err).Fatal("Failed to build runtime options")
+			}
+
+			if err := o.Run(ctx); err != nil {
+				logrus.WithError(err).Fatal("Command failed")
+			}
+
+			return nil
+		},
+
+		Args: jobrunaggregatorlib.NoArgs,
+	}
+
+	f.BindFlags(cmd.Flags())
+
+	return cmd
+}
+
+// Validate checks to see if the user-input is likely to produce functional runtime options
+func (f *JobRunsTestCaseAnalyzerFlags) Validate() error {
+	if len(f.WorkingDir) == 0 {
+		return fmt.Errorf("missing --working-dir: like -install-analyzer-working-dir")
+	}
+	if _, err := time.Parse(kubeTimeSerializationLayout, f.EstimatedJobStartTimeString); err != nil {
+		return err
+	}
+	if err := f.DataCoordinates.Validate(); err != nil {
+		return err
+	}
+	if err := f.Authentication.Validate(); err != nil {
+		return err
+	}
+	if len(f.PayloadTag) > 0 && len(f.PayloadInvocationID) > 0 {
+		return fmt.Errorf("cannot specify both --payload-tag and --payload-invocation-id")
+	}
+	if len(f.PayloadTag) == 0 && len(f.PayloadInvocationID) == 0 {
+		return fmt.Errorf("exactly one of --payload-tag or --payload-invocation-id must be specified")
+	}
+	if len(f.PayloadInvocationID) > 0 && len(f.JobGcsPrefixes) == 0 {
+		return fmt.Errorf("if --payload-invocation-id is specified, you must specify --explicit-gcs-prefixes")
+	}
+
+	bigQueryClient, err := f.Authentication.NewBigQueryClient(context.Background(), f.DataCoordinates.ProjectID)
```

Comment:
this doesn't belong in a validate step.  You can do this in `ToOptions` if you like.

Reply:
The clients are used to dynamically get all networks and platforms for validation purposes. Also, ToOptions is done after Validate.